core/src/test/java/org/apache/accumulo/core/data/KeyTest.java

```diff
@@ -79,6 +79,144 @@ public void testCopyData() {
     assertEquals(k1, k2);
 
   }
+  
+  @Test
+  public void testCopyDataWithByteArrayConstructors() {
+    byte[] row = "r".getBytes();
+    byte[] cf = "cf".getBytes();
+    byte[] cq = "cq".getBytes();
+    byte[] cv = "cv".getBytes();
+
+    Key kRow = new Key(row);
+    Key kRowcolFam = new Key(row, cf);
+    Key kRowcolFamColQual = new Key(row, cf, cq);
+    Key kRowcolFamColQualColVis = new Key(row, cf, cq, cv);
+    Key kRowcolFamColQualColVisTimeStamp = new Key(row, cf, cq, cv, 5L);
+
+    // test row constructor
+    assertNotSame(row, kRow.getRowBytes());
```

Comment:
Since these can be not the same for a variety of reasons, we probably want to check they are both not the same, and yet still equal.

Reply:
I added an equals method for the byte[] to show they are equal even when copied.